test/BasicVectorTest.template.ts

```diff
@@ -94,3 +93,48 @@ export const AddressMosaicIdTester = <T extends Network>(
         );
     });
 };
+
+export const CipherVectorTester = (testCipherVectorFile: string): void => {
+    describe('cipher - test vector', () => {
+        tester.run(
+            testCipherVectorFile,
+            (item: { privateKey: string; otherPublicKey: string; tag: string; iv: string; cipherText: string; clearText: string }) => {
+                // Act:
+                const encoded = encode(
+                    Key.createFromHex(item.privateKey),
+                    Key.createFromHex(item.otherPublicKey),
+                    Converter.hexToUint8(item.clearText),
+                    Converter.hexToUint8(item.iv),
+                );
+                // Assert:
+                const message = ` from ${item.clearText}`;
+                expect(Converter.uint8ToHex(encoded), `cipher ${message}`).equal(`${item.tag}${item.iv}${item.cipherText}`);
+            },
+            'cipher test',
+        );
+    });
+};
+
+export const DeriveVectorTester = (testCipherVectorFile: string): void => {
+    describe('Derive - test vector', () => {
+        tester.run(
+            testCipherVectorFile,
+            (item: { privateKey: string; otherPublicKey: string; scalarMulResult: string; sharedKey: string }) => {
+                // Act:
+                const sharedSecret = deriveSharedSecret(
+                    Key.createFromHex(item.privateKey).toBytes(),
+                    Key.createFromHex(item.otherPublicKey).toBytes(),
+                );
+                const sharedKey = deriveSharedKey(
+                    Key.createFromHex(item.privateKey).toBytes(),
+                    Key.createFromHex(item.otherPublicKey).toBytes(),
+                );
+                // Assert:
+                const message = ` from ${item.privateKey}`;
+                expect(Converter.uint8ToHex(sharedSecret), `ScalarMulResult ${message}`).equal(item.scalarMulResult);
+                expect(Converter.uint8ToHex(sharedKey), `SharedKey ${message}`).equal(item.sharedKey);
```

Comment:
mul = mult?

Reply:
![image](https://user-images.githubusercontent.com/4384931/132319044-fe4475fa-0c07-4651-ae73-2d51c1c639ba.png)

Yeah, the method name is using the `mult`, but the vector json key is using `mul` I was trying to make it consistent